spec/prog/postgres/postgres_server_nexus_spec.rb

```diff
@@ -100,142 +66,194 @@
-      expect(nx.strand).to receive(:label).and_return("destroy").at_least(:once)
+      nx.incr_destroy
+      postgres_resource.strand.update(label: "destroy")
+      server_strand.update(label: "destroy")
       expect { nx.before_run }.not_to hop("destroy")
     end
 
     it "does not hop to destroy if already in the wait_children_destroy state" do
-      expect(nx).to receive(:when_destroy_set?).and_yield
-      expect(resource).to receive(:strand).and_return(nil)
-      expect(nx.strand).to receive(:label).and_return("wait_children_destroy").at_least(:once)
+      nx.incr_destroy
+      postgres_resource.strand.update(label: "destroy")
+      server_strand.update(label: "wait_children_destroy")
       expect { nx.before_run }.not_to hop("destroy")
     end
 
     it "does not hop to destroy if already in the destroy_vm_and_pg state" do
-      expect(nx).to receive(:when_destroy_set?).and_yield
-      expect(resource).to receive(:strand).and_return(nil)
-      expect(nx.strand).to receive(:label).and_return("destroy_vm_and_pg").at_least(:once)
+      nx.incr_destroy
+      postgres_resource.strand.update(label: "destroy")
+      server_strand.update(label: "destroy_vm_and_pg")
       expect { nx.before_run }.not_to hop("destroy")
     end
 
     it "cancels the destroy if the server is picked up for take over" do
-      expect(nx).to receive(:when_destroy_set?).and_yield
-      expect(resource).to receive(:strand).and_return(instance_double(Strand, label: "wait"))
-      expect(postgres_server).to receive(:taking_over?).and_return(true)
-      expect(nx).to receive(:decr_destroy)
+      nx.incr_destroy
+      # Resource strand is "wait" (not destroying), server is taking over
+      server_strand.update(label: "taking_over")
       expect { nx.before_run }.not_to hop("destroy")
+      expect(Semaphore.where(strand_id: postgres_server.id, name: "destroy").count).to eq(0)
     end
 
     it "pops additional operations from stack" do
-      expect(nx).to receive(:when_destroy_set?).and_yield
-      expect(resource).to receive(:strand).and_return(instance_double(Strand, label: "destroy"))
-      expect(nx.strand).to receive(:label).and_return("destroy").at_least(:once)
-      expect(nx.strand.stack).to receive(:count).and_return(2)
-      expect { nx.before_run }.to exit({"msg" => "operation is cancelled due to the destruction of the postgres server"})
+      nx.incr_destroy
+      postgres_resource.strand.update(label: "destroy")
+      # Set up a stack with proper back-link (simulating a pushed frame)
+      server_strand.update(
+        label: "destroy",
+        stack: [
+          {"subject_id" => postgres_server.id, "link" => ["Postgres::PostgresServerNexus", "wait"]},
+          {}
+        ]
+      )
+      # Create fresh nexus so frame memoization picks up the new stack
+      fresh_nx = described_class.new(server_strand.reload)
+      expect { fresh_nx.before_run }.to hop("wait")
```

Comment:
You don't need to create a separate nexus object if you want to reference it until after modifying `server_strand`:

```suggestion
      nx.incr_destroy
      expect { nx.before_run }.to hop("wait")
```